### ISSUE TITLE
Solr error when uploading notebook

### ISSUE THREAD

**Issue by defeo** [nbgallery/nbgallery#28]
I'm getting this error when I try to upload a notebook

```
solr_1       | 2018-07-05 00:44:38.417 ERROR (qtp1521083627-25) [   x:default] o.a.s.h.RequestHandlerBase org.apache.solr.common.SolrException: ERROR: [doc=Notebook 1] Error adding field 'updated_at_d'='2018-07-05T00:44:37Z' msg=For input string: "2018-07-05T00:44:37Z"
solr_1       | 	at org.apache.solr.update.DocumentBuilder.toDocument(DocumentBuilder.java:208)
solr_1       | 	at org.apache.solr.update.AddUpdateCommand.getLuceneDocument(AddUpdateCommand.java:101)
solr_1       | 	at org.apache.solr.update.DirectUpdateHandler2.updateDocument(DirectUpdateHandler2.java:963)
solr_1       | 	at org.apache.solr.update.DirectUpdateHandler2.updateDocOrDocValues(DirectUpdateHandler2.java:954)
solr_1       | 	at org.apache.solr.update.DirectUpdateHandler2.doNormalUpdate(DirectUpdateHandler2.java:334)
solr_1       | 	at org.apache.solr.update.DirectUpdateHandler2.addDoc0(DirectUpdateHandler2.java:271)
solr_1       | 	at org.apache.solr.update.DirectUpdateHandler2.addDoc(DirectUpdateHandler2.java:221)
solr_1       | 	at org.apache.solr.update.processor.RunUpdateProcessor.processAdd(RunUpdateProcessorFactory.java:67)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.DistributedUpdateProcessor.doLocalAdd(DistributedUpdateProcessor.java:950)
solr_1       | 	at org.apache.solr.update.processor.DistributedUpdateProcessor.versionAdd(DistributedUpdateProcessor.java:1163)
solr_1       | 	at org.apache.solr.update.processor.DistributedUpdateProcessor.processAdd(DistributedUpdateProcessor.java:633)
solr_1       | 	at org.apache.solr.update.processor.LogUpdateProcessorFactory$LogUpdateProcessor.processAdd(LogUpdateProcessorFactory.java:103)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.AddSchemaFieldsUpdateProcessorFactory$AddSchemaFieldsUpdateProcessor.processAdd(AddSchemaFieldsUpdateProcessorFactory.java:475)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.FieldMutatingUpdateProcessor.processAdd(FieldMutatingUpdateProcessor.java:118)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.FieldMutatingUpdateProcessor.processAdd(FieldMutatingUpdateProcessor.java:118)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.FieldMutatingUpdateProcessor.processAdd(FieldMutatingUpdateProcessor.java:118)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.FieldMutatingUpdateProcessor.processAdd(FieldMutatingUpdateProcessor.java:118)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.FieldNameMutatingUpdateProcessorFactory$1.processAdd(FieldNameMutatingUpdateProcessorFactory.java:75)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.FieldMutatingUpdateProcessor.processAdd(FieldMutatingUpdateProcessor.java:118)
solr_1       | 	at org.apache.solr.update.processor.UpdateRequestProcessor.processAdd(UpdateRequestProcessor.java:55)
solr_1       | 	at org.apache.solr.update.processor.AbstractDefaultValueUpdateProcessorFactory$DefaultValueUpdateProcessor.processAdd(AbstractDefaultValueUpdateProcessorFactory.java:92)
solr_1       | 	at org.apache.solr.handler.loader.XMLLoader.processUpdate(XMLLoader.java:261)
solr_1       | 	at org.apache.solr.handler.loader.XMLLoader.load(XMLLoader.java:188)
solr_1       | 	at org.apache.solr.handler.UpdateRequestHandler$1.load(UpdateRequestHandler.java:97)
solr_1       | 	at org.apache.solr.handler.ContentStreamHandlerBase.handleRequestBody(ContentStreamHandlerBase.java:68)
solr_1       | 	at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:195)
solr_1       | 	at org.apache.solr.core.SolrCore.execute(SolrCore.java:2503)
solr_1       | 	at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:711)
solr_1       | 	at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:517)
solr_1       | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:384)
solr_1       | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:330)
solr_1       | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1629)
solr_1       | 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
solr_1       | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
solr_1       | 	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
solr_1       | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
solr_1       | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
solr_1       | 	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
solr_1       | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
solr_1       | 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
solr_1       | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
solr_1       | 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
solr_1       | 	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
solr_1       | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
solr_1       | 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
solr_1       | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
solr_1       | 	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:219)
solr_1       | 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
solr_1       | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
solr_1       | 	at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:335)
solr_1       | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
solr_1       | 	at org.eclipse.jetty.server.Server.handle(Server.java:530)
solr_1       | 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:347)
solr_1       | 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:256)
solr_1       | 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
solr_1       | 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
solr_1       | 	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
solr_1       | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
solr_1       | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
solr_1       | 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
solr_1       | 	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
solr_1       | 	at java.base/java.lang.Thread.run(Thread.java:844)
solr_1       | Caused by: java.lang.NumberFormatException: For input string: "2018-07-05T00:44:37Z"
solr_1       | 	at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
solr_1       | 	at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
solr_1       | 	at java.base/java.lang.Double.parseDouble(Double.java:543)
solr_1       | 	at org.apache.solr.schema.DoublePointField.createField(DoublePointField.java:156)
solr_1       | 	at org.apache.solr.schema.PointField.createFields(PointField.java:250)
solr_1       | 	at org.apache.solr.update.DocumentBuilder.addField(DocumentBuilder.java:66)
solr_1       | 	at org.apache.solr.update.DocumentBuilder.toDocument(DocumentBuilder.java:159)
solr_1       | 	... 69 more
```

I'm using the docker-compose configuration recently added by #24.

**Comment by jfleck1** [comment]
Hi defeo. It looks like a date format issue and likely a mismatch of the format your notebook is using (2018-07-05T00:44:37Z) and the format solr wants to see. Or possibly it's trying to set a  DoublePointField in solr to the date string. If you can, post the notebook you're trying to add. It's also possible that the solr schema config is off from the notebook schema.

You might also try looking into the solr's config in nbgallery/docker/config/solr/. That directory is volume mapped into the solr container with [line 35 in docker-compose.yml](https://github.com/jfleck1/nbgallery/blob/ba18bd90b75207cfa42915e3bf08c78535019f26/docker-compose.yml#L35). Nothing really jumped out for me. But you can [customize your solr config](https://lucene.apache.org/solr/guide/7_0/solr-configuration-files.html) there.

**Comment by phungc** [comment]
You need to update dynamicField in ./docker/config/solr/cores/default/managed-schema to match Sunspot type definition and restart Solr.

    <dynamicField name="*_e"  type="pdouble" indexed="true"  stored="true"/>
    <dynamicField name="*_es" type="pdoubles" indexed="true"  stored="true"/>
    <dynamicField name="*_d"  type="pdate"    indexed="true"  stored="true"/>
    <dynamicField name="*_ds" type="pdate"    indexed="true"  stored="true" multiValued="true"/>

**Comment by psychemedia** [comment]
I'm also getting an upload error when trying to use the default [docker-compose file](https://github.com/nbgallery/nbgallery/blob/master/docker-compose.yml) and the settings copied from https://github.com/nbgallery/nbgallery/tree/master/docker/config

**Comment by jeffyjefflabs** [comment]
What error message are you getting?

**Comment by psychemedia** [comment]
In the user dialogue, a pop=up *"Looks like we had an error - sorry!"*  and a banner in the upload dialogue: "Error with notebook".

Solr error:

```
RequestHandlerBase
org.apache.solr.common.SolrException: ERROR: [doc=Notebook 1] Error adding field 'updated_at_d'='2018-12-07T12:39:28Z' msg=For input string: "2018-12-07T12:39:28Z"
```

**Comment by jeffyjefflabs** [comment]
The changes in https://github.com/nbgallery/nbgallery/commit/f9b7abba5d51ea769b56c35027e7f9a9a0e85ad9 work for me out of the box (i.e. remove all containers and reset the mounted volume to what's in git), with the exception of needing the permissions workaround I posted in #38.  I also had to change solrconfig.xml to enable MoreLikeThis queries, and that in turn required another schema change due to a similar error `undefined field shares_im`.  But I can now upload a notebook and it immediately renders with the More Like This section populated.

If you start from scratch and still get errors, let us know, but I think this is fixed.

On a side note, I tried just copying the schema.xml from sunspot, but the nbgallery container wouldn't start up.
